### PR TITLE
Add KMS and VPC support

### PR DESCRIPTION
**Pull request by tleguijt**

### CODE REVIEW

File: terraform/secret/main.tf [hunk 1]
```diff
@@ -11,7 +11,9 @@ data "aws_lambda_function" "commercetools_token_refresher" {
 }
 
 resource "aws_secretsmanager_secret" "commercetools_client" {
-  name = "${var.name}/commercetools-client"
+  # WARNING: iam policy refefrences "/commercetools-client" to grant access
```
Comment (davidweterings) [inline terraform/secret/main.tf:14]: typo refefrences -> references

File: terraform/secret/main.tf [hunk 2]
```diff
@@ -20,7 +22,9 @@ resource "aws_secretsmanager_secret_version" "commercetools_client" {
 }
 
 resource "aws_secretsmanager_secret" "ct_access_token" {
-  name = "${var.name}/ct-access-token"
+  # WARNING: iam policy refefrences "/ct-access-token" to grant access
```
Comment (davidweterings) [inline terraform/secret/main.tf:25]: same typo

File: terraform/iam.tf [hunk 3]
```diff
@@ -34,11 +38,11 @@ data "aws_iam_policy_document" "lambda_policy" {
       "arn:aws:secretsmanager:${local.aws_region_name}:${local.aws_account_id}:secret:*/ct-access-token-*",
     ]
 
-    condition {
-      test     = "ArnEquals"
-      variable = "secretsmanager:resource/AllowRotationLambdaArn"
-      values   = [aws_lambda_function.commercetools_token_refresher.arn]
-    }
+    # condition {
```
Comment (davidweterings) [inline terraform/iam.tf:41]: Why don't we need this one anymore? We can remove the commented code if not needed.

File: terraform/events.tf [hunk 4]
```diff
@@ -71,3 +115,19 @@ resource "aws_lambda_permission" "allow_cloudwatch" {
   principal     = "events.amazonaws.com"
   source_arn    = aws_cloudwatch_event_rule.scope_change.arn
 }
+
+resource "aws_security_group" "lambda" {
```
Comment (davidweterings) [inline terraform/events.tf:119]: If we have no VPC we don't need to create this security group?

File: terraform/iam.tf (hunk 3 above)
Comment (tleguijt) [reply terraform/iam.tf:41]: hm removed it because a cyclic dependency. Let me double check

**Pull request merged**